lib/compress.js

```diff
@@ -1117,10 +1134,11 @@ merge(Compressor.prototype, {
             throw def;
         });
         def(AST_Dot, function(compressor){
-            if (compressor.option("unsafe") && this.property == "length") {
-                var str = ev(this.expression, compressor);
-                if (typeof str == "string")
-                    return str.length;
+            if (compressor.option("unsafe")) {
+                var val = ev(this.expression, compressor);
+                if (HOP(val, this.property)) {
```

Comment:
Hmm, your example seems to work over here: 😓 
```
C:\UglifyJS2>echo a = null.p; | node bin\uglifyjs -c unsafe
a=null.p;


C:\UglifyJS2>node
> var minify = require('uglify-js').minify;
undefined
> minify('a = null.p;', { fromString: true, compress: { unsafe: true } });
{ code: 'a=null.p;', map: null }
```

Regardless, do you suggest I follow [this line](https://github.com/mishoo/UglifyJS2/blob/master/lib/compress.js#L2621) to fix this up?